internal/cachecontroller/cache_controller_test.go

```diff
@@ -85,149 +88,209 @@ func TestInMemoryCacheController_findChangesAndInvalidate(t *testing.T) {
 	t.Cleanup(func() {
 		goleak.VerifyNone(t)
 	})
-	type readChangesResponse struct {
-		err     error
-		changes []*openfgav1.TupleChange
-	}
 
 	tests := []struct {
-		name               string
-		storeID            string
-		continuationToken  string
-		readChangesResults *readChangesResponse
-		setCacheKeys       []string
+		name     string
+		storeID  string
+		setMocks func(cache *mocks.MockInMemoryCache[any], datastore *mocks.MockOpenFGADatastore)
 	}{
 		{
-			name:               "empty_changelog",
-			storeID:            "1",
-			readChangesResults: &readChangesResponse{err: storage.ErrNotFound},
-			setCacheKeys: []string{
-				storage.GetInvalidIteratorCacheKey("1")},
+			name:    "empty_changelog",
+			storeID: "1",
+			setMocks: func(cache *mocks.MockInMemoryCache[any], datastore *mocks.MockOpenFGADatastore) {
+				gomock.InOrder(
+					datastore.EXPECT().ReadChanges(gomock.Any(), "1", gomock.Any(), gomock.Any()).Times(1).Return(nil, "", storage.ErrNotFound),
```

Comment:
here and elsewhere: 

```suggestion
					datastore.EXPECT().ReadChanges(gomock.Any(), "1", gomock.Any(), opts).Times(1).Return(nil, "", storage.ErrNotFound),
```

this will ensure that `SortDesc: true`